core/src/program.rs

```diff
@@ -278,15 +279,18 @@ impl<EC: EvalCache> Program<EC> {
             initial_env: &Environment,
         ) -> Result<RichTerm, Error> {
             vm.reset();
-            let (rt, env) = vm
-                .eval_closure(
-                    Closure {
-                        body: t,
-                        env: current_env,
-                    },
-                    initial_env,
-                )
-                .map_err(Error::from)?;
+            let result = vm.eval_closure(
+                Closure {
+                    body: t.clone(),
+                    env: current_env,
+                },
+                initial_env,
+            );
+
+            let (rt, env) = match result {
+                Err(EvalError::MissingFieldDef { .. }) => return Ok(t),
```

Comment:
Might be worth adding a comment to say why we do that in the code (basically, your PR description)